site/app/controllers/student/SubmissionController.php

```diff
@@ -113,7 +118,14 @@ private function deleteRequest(){
         }
         $this->core->getQueries()->deleteRegradeRequest($gradeable_id, $student_id);
     }
-
+    private function changeRequestStatus(){
+        $regrade_id = $_REQUEST['regrade_id'];
+        $status = $_REQUEST['status'];
+       // $gradeable_id = (isset($_REQUEST['gradeable_id'])) ? $_REQUEST['gradeable_id'] : null;
+      //  $gradeable=$this->core->getQueries()->getGradeable($gradeable_id);
+      //  $gradeable->setUserViewedDate(null);
+        $this->core->getQueries()->modifyRegradeStatus($regrade_id, $status);
```

Comment:
Would highly recommend some sort of access control, make sure they are actually allowed to change the status (it's their gradeable, status is correct, etc)